sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,237 @@
+
+    See also :ref:`User guide <scaling_regression>`
+
+    """
+    def __init__(self, regressor=None, transformer=None,
+                 func=None, inverse_func=None, check_inverse=True,
+                 random_state=None):
+        self.regressor = regressor
+        self.transformer = transformer
+        self.func = func
+        self.inverse_func = inverse_func
+        self.check_inverse = check_inverse
+        self.random_state = random_state
+
+    def _fit_transformer(self, y):
+        if (self.transformer is not None and
+                (self.func is not None or self.inverse_func is not None)):
+            raise ValueError("'transformer' and functions 'func'/"
+                             "'inverse_func' cannot both be set.")
+        elif self.transformer is not None:
+            self.transformer_ = clone(self.transformer)
+        else:
+            self.transformer_ = FunctionTransformer(
+                func=self.func, inverse_func=self.inverse_func, validate=False)
+        # XXX: sample_weight is not currently passed to the
+        # transformer. However, if transformer starts using sample_weight, the
+        # code should be modified accordingly. At the time to consider the
+        # sample_prop feature, it is also a good use case to be considered.
+        self.transformer_.fit(y)
+        if self.check_inverse:
+            random_state = check_random_state(self.random_state)
+            n_subsample = min(10, y.shape[0])
+            subsample_idx = random_state.choice(range(y.shape[0]),
+                                                size=n_subsample,
+                                                replace=False)
+            if not np.allclose(
+                    y[subsample_idx],
+                    self.transformer_.inverse_transform(
+                        self.transformer_.transform(y[subsample_idx]))):
+                raise ValueError("The provided functions or transformer are"
+                                 " not strictly inverse of each other. If"
+                                 " you are sure you want to proceed regardless"
+                                 ", set 'check_inverse=False'")
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like, shape (n_samples,) optional
+            Array of weights that are assigned to individual samples.
+            If not provided, then each sample is given unit weight.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, y = check_X_y(X, y, multi_output=True, y_numeric=True)
```

Comment:
But you might as well add a test for complex input in case we do something like change check_array to require real numbers